### PR TITLE
[T3A3][F10-A3]Chia Wei Kang

### PR DESCRIPTION
**Pull request by weikangchia**

### CODE REVIEW

**Comment by weikangchia** [comment]
Ready for review

File: src/seedu/addressbook/data/address/Block.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    public static final String EXAMPLE = "123 or 123A";
+    public static final String MESSAGE_BLOCK_CONSTRAINTS =
+            "Address block need to be in this format: number only or number followed by a alphabet";
+    public static final String BLOCK_VALIDATION_REGEX = "\\d+[A-Z,a-z]?";
+    
+    private String _value;
+
+    public Block(String value) throws IllegalValueException {
+        if (!isValidBlock(value)) {
+            throw new IllegalValueException(MESSAGE_BLOCK_CONSTRAINTS);
```
Comment (chanjunweimy) [inline src/seedu/addressbook/data/address/Block.java]: good job in trying exception :)

File: src/seedu/addressbook/data/address/Block.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    @Override
+    public String toString() {
+        return _value;
+    }
+
+    /**
+     * Returns true if a given string is a valid address block.
+     */
+    public static boolean isValidBlock(String test) {
+        return test.matches(BLOCK_VALIDATION_REGEX);
```
Comment (chanjunweimy) [inline src/seedu/addressbook/data/address/Block.java]: good job in checking the validity using regex :)

**Comment by chanjunweimy** [comment]
Good job!

**Pull request closed without merging**